app/views/contexts/calculated_summary_context.py

```diff
@@ -232,14 +240,15 @@ def _format_total(
             )
 
         if answer_format["type"] == "percentage":
-            return format_percentage(total)
+            return str(format_percentage(total))
 
-        return format_number(total)
+        return str(format_number(total))
 
     @staticmethod
     def _get_calculated_question(
         calculation_question: ImmutableDict, formatted_total: str
```

Comment:
According to the doc we need to specify top level arguments for all generic types so `ImmutableDict` in this case. There are few more that need fixing in this PR.

Reply:
As discussed with @petechd - I have provided two levels of argument of generic type
`ImmutableDict[str, Union[list[dict[str, str]], str]]`

Reply:
I think it just complicates the type for a dict. What we should have is a typed dict or a model. Since `calculation_question` comes from the schema, it will eventually have a model, so I think it was fine as it was, but happy keep it and update when we have a model.

Reply:
👍